### ISSUE TITLE
Pipeline consensus sequence accuracy and Phred scores

### ISSUE THREAD

**Issue by daov94** [benjjneb/dada2#575]
Greetings!

I am very new to bioinformatics so please bear with me and thanks in advance for your time and assistance!  We are trying to test different parameters for our DADA2 pipeline and compare the output results of each parameter trial in order to determine the most optimal parameters for both accuracy as well as final percentage of consensus sequences remaining after demultiplexing, filtering, merging, and chimera removal.  We are able to identify the percentage of consensus sequences left after processing, but we don’t know how accurate or certain these final sequences actually are.

Is there a way to take any input sample, process it via the pipeline and some set of parameters, generate average phred scores for all of the generated consensus sequences for that input sample, and then export the tabulated scores to compare the overall accuracy of the input parameter chosen?

So if I have 54,000 reads in a single soil sample, I want to generate a single average Phred score for the resultant 15,000 consensus sequences for that single soil sample, which should indicate the accuracy of the resultant sequences after the pipeline (and thus obtain for example, an average Phred score of 52 for all 15,000 final sequences generated using that one particular set of parameters).  I want to then compare that average Phred score of 52 for that sample and one particular set of parameters, to other averaged Phred scores (for example, to Phred scores of 47, 54, and 49) calculated for that same sample but with different sets of parameters.  With this I want to gauge how the different pipeline parameters affect the call accuracy of the generated consensus sequences overall.  Is there a way to do this in DADA2?

Thanks so much!

**Comment by benjjneb** [comment]
I'm not completely understanding what you are trying to do here. You want quality scores associated with the output sequences? I'm not sure that would be very informative about the accuracy. Technically you can get them from the `dada-class` objects (see `dd$quality`) but if you're going paired-end sequencing the merged sequences won't have quality scores attached.

Do you have any test samples, like mock communities, that were included in the run? That might be an easier way to get a sense of things. 15,000 ASVs (?) out of 54,000 reads sounds like a lot though, even for soil... Have you tried BLAST-ing a random sample of the output ASVs to get a sense of what they look like?

**Comment by daov94** [comment]
> You want quality scores associated with the output sequences?

Something like that yes.  We are hoping to try and get individual Phred scores (or some indicator of the quality and certainty of the base call) that indicate the certainty of the accuracy of a single ESV, then average all Phred scores for all ESV's within the sample and interpret that averaged Phred score as an indicator of how certain DADA2 is with the ESV it made.

> Do you have any test samples, like mock communities, that were included in the run? 
> Have you tried BLAST-ing a random sample of the output ASVs to get a sense of what they look like?

We unfortunately did not include any mock communities in the processing or sequencing; all of our samples are entirely environmental.  15,000 is just a completely random number I pulled out to try and describe what I was getting at, our samples have.

We have assigned taxonomy to five different parameter trials and have just begun looking through the taxonomy table to see how the community might be affected by the different DADA2 pipeline parameters we tested.  Compared to looking at Phred scores, as I think you are suggesting or implying, this is indeed probably a better indicator for how DADA2 pipeline parameters affect the community and I think we'll shift our focus toward this instead of the Phred scores.

Thanks for your time!
